test/functional/test_framework/crypto/secp256k1.py

```diff
@@ -344,3 +346,9 @@ def mul(self, a):
 
 # Precomputed table with multiples of G for fast multiplication
 FAST_G = FastGEMul(G)
+
+class TestFrameworkSecp256k1(unittest.TestCase):
+    def test_H(self):
+        H = sha256(G.to_bytes_uncompressed()).digest()
+        assert GE.lift_x(FE.from_bytes(H)) is not None
+        self.assertEqual(H.hex(), "50929b74c1a04954b78b4b6035e97a5e078a5a0f28ec96d547bfee9ace803ac0")
```

Comment:
is it possible to avoid hard coding? That kinda' defeats the purpose of the test I think

Reply:
I don't think so? The purpose of adding this code is to give a precise definition of how the constant from BIP341 is calculated. The assert confirms that this code generates a point which matches the expected value from BIP341 and allows anyone else to verify that the code == constant in `src/pubkey.cpp` == value from BIP341.

Reply:
But we're not actually testing the production code this way, i.e. the test would still pass if `NUMS_H_DATA` were to change (e.g. after an accidental local search/replace or whatever).

Reply:
Right, but that's not the intent of the python code: it's merely to provide a more precise explanation of the definition of `H`. A unit test on secp256k1 seemed like the most natural fit.

Seems like what you actually want is a unit test in the C++ code to generate the value of `H` and then compare it to `XOnlyPubKey::NUMS_H`. I've added that in https://github.com/bitcoin/bitcoin/pull/30048/commits/8d4c70c054f73f5991408e2e2bed61f2c4384f2e